buildroot: add docopt.cpp_ext package

eaaltonen opened the pull request:
Add docopt.cpp library that can be used to create intuitive command line interfaces with very little code by describing the valid calls using the (POSIX.1-2017) command line utility argument syntax.

This is in preparation for adding the Command Line tool from https://github.com/OP-TEE/optee_client/pull/259 (not merged). The intent is that the CLI tool and and this library will be optional components in the build.

 I'm opening this as a PR first here, and plan to post it a bit later also to buildroot mailing list as the `docopt.cpp` package.

<!--
    If you are new to submitting pull requests to OP-TEE, then please have a
    look at the list below and tick them off before submitting the pull request.

    1. Read our contribution guidelines:
         https://optee.readthedocs.io/en/latest/general/contribute.html

    2. Read the contribution section in Notice.md and pay extra attention to the
       "Developer Certificate of Origin" in the contribution guidelines.

    3. You should run checkpatch preferably before submitting the pull request.

    4. When everything has been reviewed, you will need to squash, rebase and
       add tags like `Reviewed-by`, `Acked-by`, `Tested-by` etc. More details
       about this can also be found on the link provided above.

    NOTE: This comment will not be shown in the pull request, so no harm keeping
    it, but feel free to remove it if you like.
-->

jenswi-linaro commented:
Any plans to upstream this to buildroot?

eaaltonen commented:
Yes, planning to submit to buildroot mailing list. Thought to see first if you have any comments.

jenswi-linaro commented:
> Yes, planning to submit to buildroot mailing list. Thought to see first if you have any comments.

Looks good to me:
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`

etienne-lms reviewed (commented):
`Acked-by: Etienne Carriere <etienne.carriere@linaro.org>`

eaaltonen commented on br-ext/package/docopt.cpp_ext/Config.in:
```diff
@@ -0,0 +1,13 @@
+config BR2_PACKAGE_DOCOPT_CPP_EXT
+	bool "docopt.cpp_ext"
+	help
+	  docopt is library that lets you define a command line
```
Indeed, fixed.

eaaltonen commented:
Submitted to buildroot as "[PATCH 1/1] package/docopt.cpp: new package".
Looks like the continuing the indentation is not necessary; removed those. Also added `depends on` for libstdcpp based on the other buildroot recipes.

eaaltonen commented:
Landed on buildroot `master` as commit `43f96ced67c9a1f84d3f0edf5cd632178eb176b8` with some fixes by the committter. For example the package name was changed do `docopt-cpp`.

jforissier commented:
> Landed on buildroot `master` as commit `43f96ced67c9a1f84d3f0edf5cd632178eb176b8` with some fixes by the committter. For example the package name was changed do `docopt-cpp`.

Good! Since Buildroot has a quarterly release cycle and the latest release was 2022.08, I guess one is coming soon. I suggest we wait for it and upgrade our manifests instead of introducing this `_ext` package. Sounds like a plan?

eaaltonen commented:
We can certainly start with that plan. If I find the time to pursue the optee_client PR _before_ Buildroot has released, I can just update this PR to match with a name change.
